tests/test_optimizer_random_dag.py

```diff
@@ -116,7 +116,7 @@ def compare_optimization_results(dag: sky.Dag, minimize_cost: bool):
                                                       dag.tasks, optimizer_plan)
 
     min_objective = find_min_objective(copy_dag, minimize_cost)
-    assert objective == min_objective
+    assert abs(objective - min_objective) < 1e-6
```

Comment:
Nice catch. I think it'd be convenient to user a larger tolerance here, e.g., 1e-3.